### PR TITLE
Add "documentType" to "documents_processed" metric

### PR DESCRIPTION
**Pull request by jonmv**
@bratseth or @baldersheim please review.

### CODE REVIEW

File: docproc/src/main/java/com/yahoo/docproc/jdisc/metric/MockMetric.java [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+/**
+ * Metric implementation for tests.
+ *
+ * @author jonmv
+ */
+public class MockMetric implements Metric {
```
Comment (jonmv) [inline docproc/src/main/java/com/yahoo/docproc/jdisc/metric/MockMetric.java:14]: Should probably put this elsewhere.

File: docproc/src/main/java/com/yahoo/docproc/DocprocExecutor.java [hunk 2]
```diff
@@ -176,4 +181,15 @@ private void logProgress(Processing processing, DocumentProcessor.Progress progr
         }
         return progress;
     }
+
+    private Function<String, Metric.Context> cachedContexts(String chainDimension) {
+        Map<String, Metric.Context> contextCache = new ConcurrentHashMap<>();
+        return documentType -> contextCache.computeIfAbsent(documentType, type -> {
+            Map<String, String> dimensions = new HashMap<>(2);
+            dimensions.put("chain", chainDimension);
+            dimensions.put("documentType", type);
```
Comment (geirst) [inline docproc/src/main/java/com/yahoo/docproc/DocprocExecutor.java:190]: In all backend metrics this dimension is named "documenttype". We should use the same her I guess.

Comment (yngveaasheim) [reply docproc/src/main/java/com/yahoo/docproc/DocprocExecutor.java:190]: Geir is correct. The dimension name must be exactly "documenttype", or it cannot be used in combination with other metrics. The option would likely be to replicated "documenttype" into "documentType" for all other metrics having this already, if we conclude "documentType" is the better name.

Comment (jonmv) [reply docproc/src/main/java/com/yahoo/docproc/DocprocExecutor.java:190]: Thanhks.

**Review by bratseth — COMMENTED** [review]
What's the purpose of the "once" thing?

Otherwise, LGTM

**Comment by jonmv** [comment]
I believe that was there to avoid reporting each operation multiple times in case of async prosessing.

**Comment by bratseth** [comment]
Ah right, it was there from before. It's not worse than it was and not public so we can clean it up later, so fine.

**Review by bratseth — APPROVED** [review]

**Comment by jonmv** [comment]
🙉

**Pull request merged**